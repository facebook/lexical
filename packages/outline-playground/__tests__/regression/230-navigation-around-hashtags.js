/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import {
    initializeE2E,
    assertHTMLSnapshot,
    assertSelection,
    repeat,
  } from '../utils';

describe('Regression test #230', () => {
  initializeE2E({chromium: true, webkit: true, firefox: true}, (e2e) => {
    it(`Is able to right arrow before hashtag after inserting text node`, async () => {
      const {page} = e2e;
      await page.focus('div.editor');
      await page.keyboard.type('#foo');
      await page.waitForSelector('.editor-text-hashtag');
      await repeat(4, async () => {
        await page.keyboard.press('ArrowLeft');
      });
      await page.keyboard.type('a');
      await page.keyboard.press('Backspace');
      await page.keyboard.press('ArrowRight');
      await assertHTMLSnapshot(page);
      await assertSelection(page, {
        anchorPath: [0,1,0],
        anchorOffset: 1,
        focusPath: [0,1,0],
        focusOffset: 1,
      });
    });
  });
});
