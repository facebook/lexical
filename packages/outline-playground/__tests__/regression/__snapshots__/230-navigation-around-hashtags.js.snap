// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regression test #230 Chromium Is able to right arrow before hashtag after inserting text node 1`] = `"<p class=\\"editor-paragraph\\" dir=\\"ltr\\"><span>﻿</span><span class=\\"editor-text-hashtag\\">#foo</span></p>"`;

exports[`Regression test #230 Firefox Is able to right arrow before hashtag after inserting text node 1`] = `"<p class=\\"editor-paragraph\\" dir=\\"ltr\\"><span>﻿</span><span class=\\"editor-text-hashtag\\">#foo</span></p>"`;

exports[`Regression test #230 Webkit Is able to right arrow before hashtag after inserting text node 1`] = `"<p class=\\"editor-paragraph\\" dir=\\"ltr\\"><span>﻿</span><span class=\\"editor-text-hashtag\\">#foo</span></p>"`;
